test/libsolidity/AnalysisFramework.cpp

```diff
@@ -56,8 +56,17 @@ AnalysisFramework::parseAnalyseAndReturnError(
 
 	m_compiler.analyze();
 
+	ErrorList errors = filterErrors(m_compiler.errors(), _reportWarnings);
+	if (errors.size() > 1 && !_allowMultipleErrors)
+		BOOST_FAIL("Multiple errors found: " + formatErrors());
+
+	return make_pair(&m_compiler.ast(""), std::move(errors));
+}
+
+ErrorList AnalysisFramework::filterErrors(ErrorList const& _errorList, bool _includeWarnings) const
+{
 	ErrorList errors;
```

Comment:
Can this function be simplified to return `_errorList` if `_includeWarnings` is true and filter out all warnings otherwise?

Reply:
Ah sorry, now I understand, it also filters specific warnings.

Reply:
Added a comment in the header.